Sources/Sharing/Internal/Reference.swift

```diff
@@ -63,12 +63,6 @@ final class _BoxReference<Value>: MutableReference, Observable, Perceptible, @un
     self.value = wrappedValue
   }
 
-  deinit {
-    #if canImport(Combine)
-      subject.send(completion: .finished)
```

Comment:
We don't want publishers to complete simply because the key goes away, so instead we'll send completion from `Shared.Box` deinitialization.